context/app/static/js/components/bulkDownload/hooks.ts

```diff
@@ -0,0 +1,196 @@
+import { useCallback, useEffect } from 'react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+
+import { bulkDownloadOptionsField, bulkDownloadMetadataField } from 'js/components/bulkDownload/bulkDownloadFormFields';
+import useBulkDownloadToasts from 'js/components/bulkDownload/toastHooks';
+import { ALL_BULK_DOWNLOAD_OPTIONS } from 'js/components/bulkDownload/constants';
+import { BulkDownloadDataset, useBulkDownloadStore } from 'js/stores/useBulkDownloadStore';
+import { useSearchHits } from 'js/hooks/useSearchData';
+import { useProtectedDatasetsForm } from 'js/hooks/useProtectedDatasets';
+import { createDownloadUrl } from 'js/helpers/functions';
+import { checkAndDownloadFile, postAndDownloadFile } from 'js/helpers/download';
+import { trackEvent } from 'js/helpers/trackers';
+import { getIDsQuery } from 'js/helpers/queries';
+
+const schema = z
+  .object({
+    ...bulkDownloadOptionsField,
+    ...bulkDownloadMetadataField,
+  })
+  .partial()
+  .required({ bulkDownloadOptions: true });
+
+export interface BulkDownloadFormTypes {
+  bulkDownloadOptions: string[];
+  bulkDownloadMetadata: boolean;
+}
+
+function useBulkDownloadForm() {
+  const {
+    handleSubmit,
+    control,
+    reset,
+    setValue,
+    formState: { errors },
+    trigger,
+  } = useForm<BulkDownloadFormTypes>({
+    defaultValues: {
+      bulkDownloadOptions: [],
+      bulkDownloadMetadata: false,
+    },
+    mode: 'onChange',
+    resolver: zodResolver(schema),
+  });
+
+  return {
+    handleSubmit,
+    setValue,
+    control,
+    errors,
+    reset,
+    trigger,
+  };
+}
+
+function useBulkDownloadDialog(deselectRows?: (uuids: string[]) => void) {
+  const { isOpen, uuids, open, close, setUuids } = useBulkDownloadStore();
+  const { control, handleSubmit, errors, reset, trigger } = useBulkDownloadForm();
+  const { toastErrorDownloadFile, toastSuccessDownloadFile } = useBulkDownloadToasts();
+
+  // Fetch datasets for the selected uuids
+  const datasetQuery = {
+    query: getIDsQuery([...uuids]),
+    _source: ['hubmap_id', 'processing', 'uuid', 'files', 'processing_type'],
+    size: 1000,
+  };
+  const { searchHits, isLoading } = useSearchHits<BulkDownloadDataset>(datasetQuery);
+  const datasets = searchHits.map(({ _source }) => _source);
+
+  // Which options to show in the dialog
+  const downloadOptions = ALL_BULK_DOWNLOAD_OPTIONS.map((option) => ({
+    ...option,
+    count: datasets.filter((dataset) => option.isIncluded(dataset)).length,
```

Comment:
We could potentially push this logic into the search-api call by using a composite aggregations query with `processing` and `processing_type`. Or we could make separate requests to get the datasets for each option.

Reply:
@NickAkhmetov thoughts?